### PR TITLE
fix: Sentry-trace header incorrectly assigned in http requests

### LINKED ISSUE #1994: Propagated `sentry-trace` should include correct parent span id
### Platform

iOS

### Installed

Swift Package Manager

### Version

all

### Steps to Reproduce

Make a call from the iOS SDK to a backend.

### Expected Result

The parent of the server transaction is the span that created the request, and not the root span (`ui.load`)

### Actual Result

image

Note: This image also has an issue with what it seems to be clock drift, but the issue is that the 
image icon expanded on the top span instead of under 
image

The `http.server` transaction shows up as a child of `ui.load` (the root span, transaction)

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Instead of sending the transaction id for 'sentry-trace' header in a network request, we are now sending the correct span id related to the network span.

## :bulb: Motivation and Context

Closes #1994 

## :green_heart: How did you test it?

Unit test

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Fixes
+
+- Network request sentry-trace propagation incorrectly assign (#2167)
```
Comment (kevinrenskers) [inline CHANGELOG.md:7]: This is kind of an indecipherable sentence 😅

Comment (brustolin) [reply CHANGELOG.md:7]: Suggestions are welcome 😁

Comment (brustolin) [inline CHANGELOG.md:7]:
```suggestion
- Sentry-trace header incorrectly assign in http requests (#2167)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Fixes
+
+- Sentry-trace header incorrectly assign in http requests (#2167)
```
Comment (kevinrenskers) [inline CHANGELOG.md:7]:
```suggestion
- Sentry-trace header incorrectly assigned to http requests (#2167)
```

?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 ### Fixes
 
+- Sentry-trace header incorrectly assigned to http requests (#2167)
+
+## 7.25.1
+
+### Fixes
```
Comment (kevinrenskers) [inline CHANGELOG.md:11]: I think a merge went wrong here, we now have a duplicate entry for 7.25.1

**Review by kevinrenskers — APPROVED** [review]

**Review by philipphofmann — COMMENTED** [review]
I think we risk being flagged by the AppStore with this change. Is there really no other way?

File: Sources/Sentry/SentryNetworkTracker.m [hunk 4]
```diff
@@ -119,6 +120,18 @@ - (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
             return;
         }
 
+        SEL setCurrentRequestSelector = NSSelectorFromString(@"setCurrentRequest:");
+        if ([sessionTask currentRequest] &&
+            [sessionTask respondsToSelector:setCurrentRequestSelector]) {
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Sentry/SentryNetworkTracker.m:125]: But it does work for iOS 12 and 13. I tried both.

And the thing is, our previous solution also does not support 12 and 13. I didn't touch the baggage header injection, which means we dont have dynamic sampling for iOS 12 and 13 also.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,10 +4,14 @@
...
 
-- Use the `component` name source for SentryPerformanceTracker (#2168)
+- Sentry-trace header incorrectly assigned to http requests (#2167)
 
 ## 7.25.1
 
+### Fixes
+
+- Use the `component` name source for SentryPerformanceTracker (#2168)
+
```
Comment (brustolin) [inline CHANGELOG.md:14]:
```suggestion

- Sentry-trace header incorrectly assigned to http requests (#2167)
- Use the `component` name source for SentryPerformanceTracker (#2168)

## 7.25.1

```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,8 @@
 
 ### Fixes
 
+
```
Comment (brustolin) [inline CHANGELOG.md:7]:
```suggestion
```

File: Sources/Sentry/SentryNetworkTracker.m [hunk 7]
```diff
@@ -119,6 +120,18 @@ - (void)urlSessionTaskResume:(NSURLSessionTask *)sessionTask
             return;
         }
 
+        SEL setCurrentRequestSelector = NSSelectorFromString(@"setCurrentRequest:");
```
[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryNetworkTracker.m:123]: Then please add a comment with the information you provided here.

**Review by philipphofmann — APPROVED** [review]
LGTM, if you add the comment why this isn't calling into private APIs.

**Pull request merged**